pkg/pillar/cmd/volumemgr/handleblob.go

```diff
@@ -0,0 +1,127 @@
+package volumemgr
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+)
+
+func lookupBlobPersistImageStatus(ctx *volumemgrContext, blobSha string) *types.BlobStatus {
+
+	if blobSha == "" {
+		return nil
+	}
+	pub := ctx.pubBlobStatus
+	s, _ := pub.Get(blobSha)
+	if s == nil {
+		log.Infof("lookupBlobPersistImageStatus(%s) not found", blobSha)
+		return nil
+	}
+	status := s.(types.BlobStatus)
+	return &status
+}
+
+func blobStatusGetAll(ctx *volumemgrContext) map[string]*types.BlobStatus {
+	pub := ctx.pubBlobStatus
+	blobShaAndBlobStatus := make(map[string]*types.BlobStatus)
+	for blobSha, blobStatusInt := range pub.GetAll() {
+		blobStatus := blobStatusInt.(types.BlobStatus)
+		blobShaAndBlobStatus[blobSha] = &blobStatus
+	}
+	return blobShaAndBlobStatus
+}
+
+func publishBlobStatus(ctx *volumemgrContext, status *types.BlobStatus) {
+	log.Infof("publishBlobStatus(%s)", status.Key())
+
+	pub := ctx.pubBlobStatus
+	key := status.Key()
+	pub.Publish(key, *status)
+}
+
+func unpublishBlobStatus(ctx *volumemgrContext, status *types.BlobStatus) {
+
+	log.Infof("publishPersistImageStatus(%s)", status.Key())
+
+	pub := ctx.pubBlobStatus
+	key := status.Key()
+	st, _ := pub.Get(key)
+	if st == nil {
+		log.Errorf("unpublishBlobStatus(%s) not found", key)
+		return
+	}
+	pub.Unpublish(key)
+}
+
+//casBlobMonitor checks for blob updates periodically
+func casBlobMonitor(ctx *volumemgrContext) {
```

Comment:
Could this goroutine potentially get stuck/hung?
If so it would make sense to add the StillRunning pattern with a unique string suffix as we do for the goroutines in zedagent such as metrics, and add that .touch file to the watchdog in device-steps.sh

Reply:
Added.

Reply:
Thanks